### PR TITLE
Move sql/sqlwire into sql/driver.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

File: sql/server.go [hunk 1]
```diff
@@ -146,9 +146,7 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // exec executes the call. Any error encountered is returned; it is the
 // caller's responsibility to update the reply.
-func (s *Server) exec(call sqlwire.Call) error {
-	req := call.Args
-	resp := call.Reply
+func (s *Server) exec(req *driver.Request, resp *driver.Response) error {
```
Comment (MBerhault) [inline sql/server.go]: any reason for this being changed at the same time? we've also been switching to the xx(req)(reply, error) pattern. Should that be the case here too?

File: sql/driver/conn.go [hunk 2]
```diff
@@ -20,8 +20,6 @@ package driver
 import (
 	"database/sql/driver"
 	"fmt"
-
```
Comment (MBerhault) [inline sql/driver/conn.go]: this isn't new to this PR, just noticing it now. We're importing database/sql/driver from within the driver package. This works, but it feels dangerous and confusing.

File: sql/server.go (hunk 1 above)
Comment (petermattis) [inline sql/server.go]: This is removal of the `sqlwire.Call` struct which was only being used in this one location. I suppose I can switch to the `xx(req)(reply, error)` pattern as well.

**Comment by MBerhault** [comment]
cool. circleci picked up the missing leaktest in wire_test.go:TestDatumString.

**Comment by MBerhault** [comment]
LGTM

File: sql/server.go (hunk 1 above)
Comment (MBerhault) [inline sql/server.go]: Up to you. I think @tamird has only been doing server side for now.

Comment (tamird) [inline sql/server.go]: Yes plz

Comment (petermattis) [inline sql/server.go]: Done.

File: sql/server.go [hunk 3]
```diff
@@ -146,21 +129,21 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // exec executes the call. Any error encountered is returned; it is the
```
Comment (tamird) [inline sql/server.go]: this comment is out of date now

Comment (tschottdorf) [inline sql/server.go]: Doesn't the caller still have to set the error? Maybe s/reply/response/?

Comment (tschottdorf) [inline sql/server.go]: oh, I see.

**Pull request merged**